src/docs/asciidoc/distributed_sql.adoc

```diff
@@ -0,0 +1,211 @@
+
+
+== Distributed SQL
+
+Distributed SQL allows you to access data stored in the Hazelcast cluster
+declaratively using SQL.
+
+=== Beta Status
+
+The distributed SQL is currently in beta. The compatibility between versions
+is not guaranteed. API might change between versions without notice.
+
+=== How Distributed SQL works
+
+When an SQL statement is submitted for execution, the engine parses and
+optimizes it using Apache Calcite. The result is an executable plan that
+is cached and reused by the subsequent executions of the same statement.
+
+The plan contains a tree of query fragments. A query fragment is a tree of
+operators that could be executed on a single member independently. Child
+fragments supply data to parent fragments, possibly through a network, until
+the root fragment is reached. The root fragment returns query results to a
+user.
+
+Only `IMap` objects are supported at the moment.
+
+=== Querying IMap
+
+==== Names
+
+The engine exposes `IMap` objects as tables in the predefined `partitioned`
+schema using exact names. This schema is in the engine search path so that
+`IMap` objects could be accessed with or without the schema name.
+
+Schema and table names are case sensitive. For example, the `IMap` with the name
+`employee` can be accessed as `employee` or `partitioned.employee`, but not as
+`Employee`.
+
+==== Fields
+
+The engine reads the first local entry of the `IMap` to get the list of fields
+accessible from the SQL. The engine extracts field metadata (name, type)
+from the key and the value, and combines them into a final list of fields.
+
+Field metadata extraction logic depends on the serialization format, as described
+below.
+
+-  For Java objects (`Serializable`, `DataSerializable`, `IdentifiedDataSerializable`),
+the object is deserialized if needed and then analyzed using reflection. Only public
```

Comment:
No need to mention the implementation detail of "reflection". Non-Java developers won't know what is reflection.

Reply:
If a user `Serializable`, `DataSerializable` or `IdentifiedDataSerializable`, then there are definitely Java classes that we work with. We can't avoid mention reflection because this is how it actually works. 

Yes, it is Java-centric, but this is the most common serialization mechanism in the product currently. Java-independent approaches are `Portable` (explained below), JSON (not supported), GenericRecord (not implemented/integrated).